remote/remote.go

```diff
@@ -674,6 +688,11 @@ func (i *Image) Save(additionalNames ...string) error {
 	cfg = cfg.DeepCopy()
 
 	layers, err := i.image.Layers()
+	if len(layers) == 0 && i.addEmptyLayerOnSave {
+		empty := static.NewLayer([]byte{}, types.OCILayer)
+		i.image, err = mutate.AppendLayers(i.image, empty)
```

Comment:
Is it possible to derive the diffID of the layer? Could you call `GetLayer()` on it? That might make this testable (without having to modify the `Image` interface) per https://github.com/buildpacks/lifecycle/pull/852/files#r860932594

Reply:
In fact, that call throws the following error:  `mismatched fs layers (1) and diff ids (0)`